root/etc/e-smith/templates/etc/ssh/sshd_config/45AllowGroups2Sshd

```diff
@@ -0,0 +1,49 @@
+{
+    # Allow groups to sftp/ssh with different policies
+    #
+    # if $sssd{ShellOverrideStatus} is disabled we are in legacy (everybody is allowed with a shell access)
+    # if $sssd{ShellOverrideStatus} is enabled we allow to root and members of domain admins group
+    # if $sssd{ShellOverrideStatus} is enabled  with non empty $sshd{AllowGroups} 
+    #   we allow root, members of domain admins group and members of groups inside the AllowGroups prop
+    #
+    # members restricted to sftp (SftpRestrictedGroups) must be allowed to sshd also
+
+    sub uniq {
+        my %seen;
+        return grep { !$seen{$_}++ } @_;
+    }
+
+    # ssh doesn't accept to login if we restrict with short group name
+    # It seems to want the long group name group@domain.com
+    my $domain = $DomainName || die ('Cannot retrieve DomainName');
+    my $PermitRootLogin = $sshd{'PermitRootLogin'} || "no";
```

Comment:
Since the final response for `root` login is given by `40PermitRootLogin` fragment, I'd simplify the below code:
- remove any check on `PermitRootLogin`
- always add `root` group to `AllowGroups` option

This is just a suggestion to simplify the code, such modification is not really necessary.

Reply:
if possible I would prefer to hide the root if not allowed, else I need to test it and check to be sure

Reply:
Ok!